### PR TITLE
Test CI on Python 3.11 beta

### PR DESCRIPTION
**Pull request by dstansby**
See https://github.com/astropy/reproject/issues/297 - hopefully this will confirm or not whether building reproject works on Python 3.11

### CODE REVIEW

**Comment by astrofrog** [comment]
@dstansby could you rebase this and update it to use rc1?

File: .github/workflows/ci_workflows.yml [hunk 1]
```diff
@@ -14,17 +14,11 @@ jobs:
...
-        - macos: py310-test
-        - linux: py38-test-oldestdeps
-        - linux: py38-test
-        - linux: py39-test
-        - linux: py310-test
-        - windows: py38-test-oldestdeps
-        - windows: py39-test
-        - windows: py310-test
+        - linux: py311
+          python-version: '3.11-rc1'
```
Comment (Cadair) [inline .github/workflows/ci_workflows.yml:18]:
```suggestion
          python-version: '3.11.0-rc.1'
```

**Comment by Cadair** [comment]
Well we have a "real" error now, but it's inside compiling pyyaml :facepalm:

**Comment by Cadair** [comment]
Once https://github.com/astropy/astropy/pull/13614 is merged we should be able to come back to this.

**Comment by Cadair** [comment]
I have had to exclude sunpy from the test deps as aiohttp wont build on 3.11 yet.

**Comment by astrofrog** [comment]
The wheel builds now include Python 3.11, which is failing with the same error that aiohttp had: https://github.com/aio-libs/aiohttp/issues/6600

**Pull request merged**